docs/stable/core_extensions/iceberg/iceberg_rest_catalogs.md

```diff
@@ -0,0 +1,120 @@
+---
+layout: docu
+redirect_from: null # maybe redirect from old amazon_s3_tables and amazon_sagemaker_lakehouse docs?
+title: Iceberg Rest Catalogs
+---
+
+The `iceberg` extension supports attaching Iceberg Rest Catalogs. Before attaching an Iceberg Rest Catalog, you must install the `iceberg` extension by following the instructions located in the [overview]({% link docs/stable/core_extensions/iceberg/overview.md %}).
+
+If you are attaching to an Iceberg Rest Catalog managed by Amazon, please see the instructions for attaching to [Amazon S3 tables]({% link docs/stable/core_extensions/iceberg/amazon_s3_tables.md %}) or [Amazon Sagemaker Lakehouse]({% link docs/stable/core_extensions/iceberg/amazon_sagemaker_lakehouse.md %}).
+
+For all other Iceberg Rest Catalogs, you can follow the instructions below. Please see the [Examples](#examples) section for questionsabout specific catalogs.
+
+Most Iceberg Rest Catalogs authenticate via Oauth2. You can use the existing DuckDB secret workflow to create the oauth secret.
+
+```sql
+CREATE SECRET iceberg_secret (
+    TYPE ICEBERG,
+    CLIENT_ID '⟨admin⟩',
+    CLIENT_SECRET '⟨password⟩',
+    OAUTH2_SERVER_URI '⟨http://irc_host_url.com/v1/oauth/tokens⟩'
+);
+```
+
+If you already have a token, you can pass it directly to your `CREATE SECRET` statement
+
+```sql
+CREATE SECRET iceberg_secret (
+    TYPE ICEBERG,
+    TOKEN '⟨one_time_token⟩'
+);
+```
+
+You can attach the iceberg catalog with the following `Attach` statement.
+
+```sql
+ATTACH '⟨warehouse-name⟩' AS iceberg_catalog (
+   TYPE iceberg,
+   SECRET iceberg_secret, -- pass a specific secret name to prevent ambiguity
```

Comment:
It's not really for ambiguity, I think?
It's more that authorization options can be provided directly inside the `ATTACH` (in the same way as the "create secret" options), but for reusability/deduplication the secret can be made beforehand and reused in the attach afterwards

Reply:
Also that, but I can imagine users attach multiple catalogs that all require a different secret. I think it's best to go over all the options in a separate section